### PR TITLE
feat: Add support for creating CloudWatch log groups

### LINKED ISSUE #340: Add cloudwatch log group retention period parameter
## Is your request related to a new offering from AWS?
No

## Is your request related to a problem? Please describe.
Currently the Cloudwatch log group is automatically created with a retention period of "Never expire". This is not ideal as it leads to unexpected costs.

## Describe the solution you'd like.
In line with other `terraform-aws-rds` modules, it would be nice to specify the log group creation and retention period.

## Describe alternatives you've considered.
- Manually changing the retention period. Not infra-as-code :(
- Add a `aws_cloudwatch_log_group` resource manually outside the module. Could lead to situation where the log group has been automatically created and the resource outside the module fails.

## Additional context
It should be easy to add the functionality copy-paste from one of the other official modules to support this functionality.

### PR DESCRIPTION
**Pull request by magreenbaum**
## Description
Add cloudwatch log group management. 

## Motivation and Context
- Resolves #340

## Breaking Changes
No.

## How Has This Been Tested?
- [x] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects

- [x] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

**Review by bryantbiggs — APPROVED** [review]
This looks great - thank you!

**Comment by Rukeith** [comment]
looking forward to this PR.

File: main.tf [hunk 1]
```diff
@@ -442,7 +442,7 @@ resource "aws_db_parameter_group" "this" {
 ################################################################################
 
 resource "aws_cloudwatch_log_group" "this" {
-  for_each = toset([for log in var.enabled_cloudwatch_logs_exports : log if local.create_cluster && var.create_cloudwatch_log_group])
+  for_each = toset([for log in var.enabled_cloudwatch_logs_exports : log if local.create_cluster && var.create_cloudwatch_log_group && !var.cluster_use_name_prefix])
```
Comment (magreenbaum) [inline main.tf:445]: Updated cloudwatch log group creation based on [this feedback](https://github.com/terraform-aws-modules/terraform-aws-rds/issues/477) which is also relevant to this module.

**Comment by bryantbiggs** [comment]
CI is green so merging!

**Pull request merged**